core/src/data-generator/custom-dataset.ts

```diff
@@ -0,0 +1,228 @@
+    sourceBucket.grantRead(glueRole, props.s3Location.objectKey + '*');
+    outputBucket.grantReadWrite(glueRole, props.s3Location.objectKey + '*');
+
+    // the SSM parameter used to stored the output of the Glue job
+    const ssm = new StringParameter(this, 'JobOutputParameter', {
+      stringValue: 'minDatetime',    
+    });
+    ssm.grantWrite(glueRole);
+
+    glueRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSGlueServiceRole'));
+
+    // create a KMS key for the Glue security configureation
+    const glueKey = new Key(this, 'GlueKey', {
+      removalPolicy: RemovalPolicy.DESTROY,
+    })
+    
+    // We add a resource policy so the key can be used in Cloudwatch logs
+    glueKey.addToResourcePolicy(new PolicyStatement({
+      principals: [
+        new ServicePrincipal(`logs.${Aws.REGION}.amazonaws.com`)
+      ],
+      actions: [
+        "kms:Encrypt*",
+        "kms:Decrypt*",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:Describe*"
+      ],
+      resources: ["*"],
+      conditions: { ArnLike: { "kms:EncryptionContext:aws:logs:arn": `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:*` } },
+    }))
+
+    // the Glue job security configuration following best practices
+    const conf = new SecurityConfiguration(this, 'SecurityConfiguration', {
+      securityConfigurationName: id,
+      cloudWatchEncryption: {
+        mode: CloudWatchEncryptionMode.KMS,
+        kmsKey: glueKey,
+      },
+      jobBookmarksEncryption: {
+        mode: JobBookmarksEncryptionMode.CLIENT_SIDE_KMS,
+        kmsKey: glueKey,
+      },
+      s3Encryption: {
+        mode: S3EncryptionMode.S3_MANAGED,
+      }
+    });
+
+    // calculate the size of the Glue job based on input data size (1 + 1 DPU per 2GB)
+    var workerNum = 9;
+    if (props.approximateDataSize !== undefined && props.approximateDataSize > 16){
+      workerNum = Math.ceil(props.approximateDataSize / 2) + 1;
+    }
+
+    // Glue job to prepare the dataset
+    const glueJob = new SynchronousGlueJob(scope, 'CustomDatasetJob', {
+      executable: PreBundledPysparkJobExecutable.pythonEtl({
+        glueVersion: GlueVersion.V3_0,
+        pythonVersion: PythonVersion.THREE,
+        codePath: 'data-generator/resources/glue/custom-dataset/script.py',
+      }),
+      defaultArguments: {
+        '--s3_input_path': `s3://${props.s3Location.bucketName}/${props.s3Location.objectKey}`,
+        '--s3_output_path': `s3://${outputBucket.bucketName}/${props.s3Location.objectKey}`,
+        '--input_format': props.inputFormat,
+        '--datetime_column': props.datetimeColumn,
+        '--partition_range': props.partitionRange.toMinutes().toString(),
+        '--ssm_parameter': ssm.parameterName,
+        '--enable-auto-scaling': 'true',
+      },
+      role: glueRole,
+      workerCount: props.approximateDataSize ? workerNum : 100,
+      workerType: WorkerType.G_1X,
```

Comment:
Would it be possible to use the FLEX price type or the smaller worker recently launched?